### PR TITLE
feat: Add the creation of 'Microsoft.ApiManagement/service/loggers' and 'Microsoft.ApiManagement/service/apis/diagnostics' in Api Management Module. - `avm/res/api-management/service`

### LINKED ISSUE #1124: [AVM Module Issue]: API Management Module should add the creation of 'Microsoft.ApiManagement/service/loggers' and 'Microsoft.ApiManagement/service/apis/diagnostics'
### Check for previous/existing GitHub issues

- [X] I have checked for previous/existing GitHub issues

### Issue Type?

Feature Request

### Module Name

avm/res/api-management/service

### (Optional) Module Name if not listed above

avm/res/api-management/service

### (Optional) Module Version

_No response_

### Description

Please add the creation of `Microsoft.ApiManagement/service/loggers` and `Microsoft.ApiManagement/service/apis/diagnostics`.
For example (azd): 
https://github.com/Azure/azure-dev/blob/main/templates/common/infra/bicep/core/gateway/apim.bicep#L61
https://github.com/Azure-Samples/todo-java-mongo/blob/main/infra/app/apim-api.bicep#L56

CC: @jongio

### (Optional) Correlation Id

_No response_

### PR DESCRIPTION
**Pull request by zedy-wj**
## Description

Fixes #1124 
## Pipeline Reference

| Pipeline |
| -------- |
| [avm.res.api-management.service](https://github.com/zedy-wj/bicep-registry-modules/actions/workflows/avm.res.api-management.service.yml)         |

## Type of Change

- [ ] Update to CI Environment or utlities (Non-module effecting changes)
- [ x] Azure Verified Module updates:
  - [ ] Bugfix containing backwards compatible bug fixes, and I have NOT bumped the MAJOR or MINOR version in `version.json`:
    - [ ] Someone has opened a bug report issue, and I have included "Closes #{bug_report_issue_number}" in the PR description.
    - [ ] The bug was found by the module author, and no one has opened an issue to report it yet.
  - [ ] Feature update backwards compatible feature updates, and I have bumped the MINOR version in `version.json`.
  - [ ] Breaking changes and I have bumped the MAJOR version in `version.json`.
  - [ ] Update to documentation

## Checklist

- [ x] I'm sure there are no other open Pull Requests for the same update/change
- [ x] I have run `Set-AVMModule` locally to generate the supporting module files.
- [ x] My corresponding pipelines / checks run clean and green without any errors or warnings

@jongio - for notification.

### CODE REVIEW

File: avm/res/api-management/service/main.bicep
Comment (eriqua) [inline avm/res/api-management/service/main.bicep:1]: Since new features are introduced please update minor in version.json

**Comment by tony-box** [comment]
@zedy-wj @eriqua just a heads up--I was in the middle of developing these features in my own repo. I was almost done with them--i'm currently finalizing the deployment tests to make sure this deploys successfully.

@zedy-wj I am going to push my updates to my own development branch--if you don't mind, I'd like to work with you to compare my changes and yours so that we can align on how to best create this PR. There are some ways in which I'd prefer the modules be created but it's difficult to explain in code comments.

**Comment by jongio** [comment]
@tony-box - Thanks for doing this.  Do you have an ETA?  How can we help?

**Comment by tony-box** [comment]
@jongio I am waiting for another PR to get pushed through #1989 -- running the final validation pipelines on that right now and it should be good to go. Once that is complete I will finish the work on this one. I've got loggers fully working and diagnostics are nearly there--just a couple more issues I have to fix. 

I expect I can get the PR in by the end of this week. Thanks for your patience! I am still learning the development process for AVM. I expect I'll get quicker at it once I get these first few PRs under my belt.

**Comment by jongio** [comment]
Sounds good.  Thank you.

**Comment by tony-box** [comment]
closing this PR which will be superceeded by #2453

**Pull request closed without merging**